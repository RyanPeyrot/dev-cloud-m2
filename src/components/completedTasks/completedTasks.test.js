import React, {useContext} from "react";
import {render} from "@testing-library/react";
import CompletedTasks from "./index";
import {TasksContext} from "../../utils/contexts/taskContext";

test('No render if not any completed tasks', () => {
    const taskList = [
        { id: 1, text: 'Task 1', completed: false },
    ];

    const renderWithTasksContext = (taskList) => {
        return render(
            <TasksContext.Provider value={{ tasks: taskList }}>
                <CompletedTasks />
            </TasksContext.Provider>
        );
    };

    const completedTasksElement = renderWithTasksContext(taskList).getByRole('list');
    expect(completedTasksElement).toBeEmptyDOMElement();

})

test('Completed tasks are rendered', ()=> {
    const taskList = [
        {id: 3, text: 'Task 3', completed: true},
    ];

    const renderWithTasksContext = (taskList) => {
        return render(
            <TasksContext.Provider value={{tasks: taskList}}>
                <CompletedTasks/>
            </TasksContext.Provider>
        );
    };

    const completedTasksElement = renderWithTasksContext(taskList).getAllByRole('listitem');
    expect(completedTasksElement).toHaveLength(1)
})

test('Uncompleted tasks are not rendered', () => {
    const taskList = [
        {id: 2, text: 'Task 2', completed: false},
        {id: 3, text: 'Task 3', completed: true},
    ];

    const renderWithTasksContext = (taskList) => {
        return render(
            <TasksContext.Provider value={{tasks: taskList}}>
                <CompletedTasks/>
            </TasksContext.Provider>
        );
    };

    const completedTasksElement = renderWithTasksContext(taskList).getAllByRole('listitem');
    expect(completedTasksElement).toHaveLength(0) //1 working

})

test('Render components snapshot', () => {
    const taskList = [
        {id: 2, text: 'Task 2', completed: false},
        {id: 3, text: 'Task 3', completed: true},
    ];

 const {asFragment} = render(
     <TasksContext.Provider value={{tasks: taskList}}>
        <CompletedTasks/>
     </TasksContext.Provider>
 );
 expect(asFragment()).toMatchSnapshot()
})
