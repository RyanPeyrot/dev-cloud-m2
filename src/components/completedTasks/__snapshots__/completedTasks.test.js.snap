// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Tâches terminées
    </h2>
    <ul>
      <li>
        Task 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
